rasa_nlu/model.py

```diff
@@ -0,0 +1,272 @@
+
+        timestamp = datetime.datetime.now().strftime('%Y%m%d-%H%M%S')
+        metadata = {
+            "language": self.config["language"],
+            "pipeline": [component.name for component in self.pipeline],
+        }
+
+        if create_unique_subfolder:
+            dir_name = os.path.join(path, "model_" + timestamp)
+            os.makedirs(dir_name)
+        else:
+            dir_name = path
+
+        metadata.update(self.training_data.persist(dir_name))
+
+        for component in self.pipeline:
+            update = component.persist(dir_name)
+            if update:
+                metadata.update(update)
+
+        Metadata(metadata, dir_name).persist(dir_name)
+
+        if persistor is not None:
+            persistor.send_tar_to_s3(dir_name)
+        logging.info("Successfully saved model into '{}'".format(dir_name))
+        return dir_name
+
+
+class Interpreter(object):
+    """Use a trained pipeline of components to parse text messages"""
+
+    # Defines all attributes (and their default values) that will be returned by `parse`
+    default_output_attributes = {"intent": None, "entities": [], "text": ""}
+
+    @staticmethod
+    def load(meta, rasa_config):
+        # type: (Metadata, RasaNLUConfig) -> Interpreter
+        """Load a stored model and its components defined by the provided metadata."""
+        from rasa_nlu.registry import load_component_by_name
+
+        context = {"model_dir": meta.model_dir}
+
+        config = dict(rasa_config.items())
+        config.update(meta.metadata)
+
+        pipeline = []
+
+        for component_name in meta.pipeline:
+            try:
+                component = load_component_by_name(component_name, context, config)
+            except rasa_nlu.components.MissingArgumentError as e:
+                raise Exception("Failed to create/load component '{}'. {}".format(component_name, e.message))
+            try:
+                rasa_nlu.components.init_component(component, context, config)
+                pipeline.append(component)
+            except rasa_nlu.components.MissingArgumentError as e:
+                raise Exception("Failed to initialize component '{}'. {}".format(component.name, e.message))
+
+        return Interpreter(pipeline, context, config, meta)
+
+    def __init__(self, pipeline, context, config, meta=None):
+        # type: ([Component], dict, dict, Optional[Metadata]) -> None
+
+        self.pipeline = pipeline
+        self.context = context
+        self.config = config
+        self.meta = meta
+        self.output_attributes = [output for component in pipeline for output in component.output_provides]
+
+    def parse(self, text):
+        # type: (basestring) -> dict
+        """Parse the input text, classify it and return an object containing its intent and entities."""
+
+        if type(text) is str:
+            text = unicode(text, "utf-8")
+
+        current_context = self.context.copy()
+
+        current_context.update({
```

Comment:
so it contains both the arguments to be passed to each component, and the output from each component as well right?